docs/reference/asyncapi-rules.md

```diff
@@ -0,0 +1,325 @@
+
+    ## Authentication
+
+    This API uses OAuth2 and tokens can be requested from [Dev Portal: Tokens](https://example.org/developers/tokens).
+```
+
+### asyncapi2-info-license-url
+
+Mentioning a license is only useful if people know what the license means, so add a link to the full text for those who need it.
+
+**Recommended:** No
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  license:
+    name: MIT
+    url: https://www.tldrlegal.com/l/mit
+```
+
+### asyncapi2-info-license
+
+The `info` object should have a `license` key.
+
+It can be hard to pick a license, so if you don't have a lawyer around you can use [TLDRLegal](https://tldrlegal.com/) and [Choose a License](https://choosealicense.com/) to help give you an idea.
+
+How useful this is in court is not entirely known, but having a license is better than not having a license.
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  license:
+    name: MIT
+```
+
+### asyncapi2-operation-description
+
+Operation objects should have a description.
+
+**Recommended:** Yes
+
+### asyncapi2-operation-operationId
+
+This operation ID is essentially a reference for the operation, which can be used to visually suggest a connection to other operations. This is like some theoretical static HATEOAS-style referencing, but it's also used for the URL in some documentation systems.
+
+Make the value `lower-hyphen-case`, and try and think of a name for the action which does not relate to the HTTP message. Base it off the actual action being performed. `create-polygon`? `search-by-polygon`? `filter-companies`?
+
+**Recommended:** Yes
+
+### asyncapi2-parameter-description
+
+Parameter objects should have a `description`.
+
+**Recommended:** No
+
+### asyncapi2-payload-default
+
+`default` objects should be valid against the `payload` they decorate.
+
+**Recommended:** Yes
+
+### asyncapi2-payload-example
+
+`example` objects should be valid against the `payload` they decorate.
+
+**Recommended:** Yes
+
+### asyncapi2-payload-examples
+
+Values of the `examples` array should be valid against the `payload` they decorate.
+
+**Recommended:** Yes
+
+### asyncapi2-payload-unsupported-schemaFormat
+
+Spectral currently only supports payload validation against the default AsyncAPI schema.
+
+This rule will notify you when a payload is decorated with a `schemaFormat` property. Spectral won't be able to validate it and potential issues in this payload will go undetected by the `asyncapi2-payload-*` rules.
+
+**Recommended:** Yes
+
+### asyncapi2-payload
+
+When `schemaFormat` is undefined, the `payload` object should be valid against the AsyncAPI 2 Schema Object definition.
+
+**Recommended:** Yes
+
+### asyncapi2-schema-default
+
+`default` objects should be valid against the `schema` they decorate.
+
+**Recommended:** Yes
+
+### asyncapi2-schema-example
+
+`example` objects should be valid against the `schema` they decorate.
```

Comment:
Same as above. This one doesn't exist on AsyncAPI.

Reply:
@fmvilas Dropped.